float64 rover_x
float64 rover_y